python/spacewalk/satellite_tools/appstreams.py

```diff
@@ -148,10 +150,19 @@ def _index_modulemd(self):
         """Indexes the Modulemd file."""
         idx = Modulemd.ModuleIndex.new()
         try:
-            idx.update_from_file(self.modulemd_file, False)
+            # The following returns a tuple in the following format: (success, [failures])
+            result = idx.update_from_file(self.modulemd_file, False)
+            if not result[0]:
+                # Parsing error in a YAML subdocument
+                raise ModuleMdIndexingError(
+                    "An error occurred while indexing a module entry:",
+                    [f.get_gerror().message for f in result[1]],
+                )
         except gi.repository.GLib.GError as e:
+            # General parsing error in file
             raise ModuleMdIndexingError(
-                f"An error occurred while indexing the module metadata from file '{self.modulemd_file}'."
+                "An error occurred while indexing the module metadata from file:",
```

Comment:
The filename is not mentioned in `e.message`, but it's always printed a few lines above (see UI diff) and it's a long name (sha256 checksum in the name) so I didn't want to print it a second time to prevent a very long message.